amt.go

```diff
@@ -30,42 +31,73 @@ type Root struct {
 	store cbor.IpldStore
 }
 
-func NewAMT(bs cbor.IpldStore) *Root {
+func NewAMT(bs cbor.IpldStore, opts ...Option) (*Root, error) {
```

Comment:
This is going to require changing all invocations of `MakeEmptyArray`. The alternative is to simply panic on bad options (which isn't totally unreasonable, but is a bit unfortunate).

Reply:
Thanks. Absolutely please force all callers to acknowledge the change.